File: crates/marla_core/src/entities/page.rs
use std::{collections::BTreeMap, path::PathBuf};

use super::fields::{
    page_draft::PageDraft, page_template_path::PageTemplatePath, required_string::RequiredString,
};

#[derive(Clone, Debug)]
#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
pub struct Page {
    pub title: RequiredString,
    pub path: PathBuf,
    pub content_path: PathBuf,
    pub content: RequiredString,
    pub plain_content: RequiredString,
    pub last_modified_at: chrono::DateTime<chrono::Utc>,
    pub created_at: chrono::DateTime<chrono::Utc>,
    pub words: usize,
    pub description: Option<String>,
    pub tags: Vec<String>,
    pub slug: Option<String>,
    pub aliases: Vec<String>,
    pub template_path: PageTemplatePath,
    pub authors: Vec<String>,
    pub draft: PageDraft,
    pub extra: BTreeMap<String, String>,
    pub lang: Option<String>,
}

File: pkg/core/entities/page.go
package entities

import (
	"net/url"
	"time"

	"github.com/marlaone/marla/pkg/core/fields"
)

type Page struct {
	Parent   *Page
	Children []*Page

	Title          fields.RequiredString
	Path           *url.URL
	ContentPath    fields.Path
	Content        []byte
	PlainContent   []byte
	CreatedAt      time.Time
	LastModifiedAt *time.Time
	Words          uint
	Description    string
	Tags           []string
	Slug           string
	Aliases        []string
	Template       fields.Path
	Authors        []string
	IsDraft        bool
	Extra          map[string]any
	Language       fields.RequiredString
}
